src/main/java/fi/aalto/cs/apluscourses/model/APlusAuthentication.java

```diff
@@ -0,0 +1,26 @@
+package fi.aalto.cs.apluscourses.model;
+
+import java.util.Arrays;
+import org.jetbrains.annotations.NotNull;
+
+public class APlusAuthentication {
+  @NotNull
+  private char[] token;
+
+  /**
+   * Initializes an instance with the given token. Note, that the given token array is not cleared
+   * or overwritten.
+   */
+  public APlusAuthentication(@NotNull char[] token) {
+    this.token = token.clone();
+  }
+
+  @NotNull
+  public char[] getToken() {
+    return token;
+  }
```

Comment:
Remark: Let's see later what kind of interface this class should have. It could be possible that this class wouldn't actually have a getter but instead it would write the token straight to a network stream. But let's keep this as-is for now.

Reply:
Agreed :+1:  that would also make it easy to switch to username & password later, since this would essentially just be a black box that modifies the HTTP headers as needed